app/businessrates/authorisation/repositories/AccountsCache.scala

```diff
@@ -72,13 +73,20 @@ class AccountsMongoCache @Inject()(db: MongoComponent)(implicit ec: ExecutionCon
     collection.findOneAndDelete(equal("_id", sessionId)).toFuture().map { _ =>
       ()
     }
-  override def updateCreatedAtTimestampById(record: Seq[Record]): Future[Long] =
+  override def updateCreatedAtTimestampById(ids: Seq[String]): Future[Long] =
     collection
-      .updateMany(filter = in("_id", record.map(_._id): _*), update = set("createdAt", LocalDateTime.now()))
+      .updateMany(filter = in("_id", ids: _*), update = set("createdAt", LocalDateTime.now()))
       .toFuture()
       .map(_.getModifiedCount)
-  override def getRecordsWithIncorrectTimestamp: Future[Seq[Record]] =
-    collection.find(`type`("createdAt", BsonType.STRING)).limit(10000).toFuture()
+  override def getRecordsWithIncorrectTimestamp: Future[Seq[String]] = {
+    val longReads: Reads[String] = (__ \ "_id").read[String]
+    collection
+      .find[BsonValue](`type`("createdAt", BsonType.STRING))
+      .projection(Projections.exclude("createdAt", "data"))
```

Comment:
We must remove this job after a while tho. It's not future-proof. As soon as someone adds another attribute to the account object stored in the cache, this projection will retrieve the new field I think. But the data there hasn't changed in like a decade and this will work fine for the time we need it to